_includes/specials/taproot/en/06-multisignatures.md

```diff
@@ -56,11 +56,17 @@ family:
 
 - **MuSig2**, also simple to implement.  It eliminates one round of
   communication and allows another round to be combined with key
-  exchange.  That allows using a somewhat similar signing
+  exchange.  That can allow using a somewhat similar signing
   process to what we use today with script-based multisig.  This does
-  require storing extra data and ensuring your signing software or
+  require storing extra data and being very careful about ensuring your signing software or
   hardware can't be tricked into unknowingly repeating part of the
-  signing session.
+  signing session.  Proper state management that works even when
+  software and hardware fails is a challenging problem, so it may only be
+  advisible to use the most optimized form of MuSig2 in protocols that
+  already depend on careful state management, such as LN.
```

Comment:
This doesn't seem to be quite right. You need to be just as careful with your state management if you don't share the nonces before knowing the message. The only advantage is that you have some defense in depth when you mess up the randomness (or counter). Only MuSig-DN solves the state management issue.

What was a bit problematic in the optech newsletter was the implication that a hardware wallet could share nonces and then any time later sign with these nonces. That's very hard to implement securely compared to using MuSig2 in Lightning because 1) the session state (in particular the nonces) exist only in memory and 2) the memory is assumed to be not tampered with.

Reply:
> This doesn't seem to be quite right.

Ok, dropping this whole text in favor of "We'll examine the tradeoffs in more detail in next week's *preparing for taproot* column."  We were scheduled to cover adaptor sigs next week (see #609 ) but I think the discussions from IRC about presharing versus pipelining with musig2 are interesting and useful enough to cover, and there's no better time than right after we first described musig2.